content/blog/doomsday-rule/index.md

```diff
@@ -0,0 +1,685 @@
+
+```python
+day = int(input("What day do you want? (number needed) "))
+month = int(input("What month do you want? (number needed) "))
+year = int(input("What year do you want? "))
+```
+
+We put `int` there to tell the program that whatever the input is, it's an integer not a string.
+
+Now we need the script to work out the anchor day for the century. Whilst I was making the script whenever I tried to do `century MOD 4` it would be wrong, but on Wikipedia, it has a different way of working out the anchor day which is:
+
+`5 * (Century MOD 4) MOD 7 + 2`
+
+The way you work out the century is:
+
+`year / 100`
+
+However, unless the year ended in 00, this output would be a decimal, but we need a non-decimal number. So in code we write it as:
+
+`int(year / 100)`
+
+So under the inputs we put:
+
+```python
+Century = int(year / 100)
+
+anchor = 5 * (Century % 4) % 7 + 2
+```
+(the percentage sign (%) is python's MOD)
+
+The program now can get the anchor day. The next step is for it to work out the doomsday for the year. But we have not told the program how to get the last 2 digits of the year, so we tell the script to convert it to a string, then tell it to get the 3rd and 4th digit. so this would look like this:
+
+```python
+EndTwo = str(year)
+EndTwo = EndTwo[2:4]
+```
+
+So going back to the a, b and c calculations they would look like this:
+
+```python
+A = int(int(EndTwo) / 12)
+
+B = int(EndTwo) % 12
+
+C = int(B / 4)
+```
+
+Then add them together and MOD the sum. After that you want to add the anchor. So in code this would look like:
+
+```python
+Doomsday = ((A + B + C) % 7) + anchor
+```
+
+Then just get the MOD of `Doomsday`:
+
+```python
+Doomsday = Doomsday % 7
+```
+
+So far the script looks like:
+```python
+day = int(input("What day do you want? (number needed) "))
+month = int(input("What month do you want? (number needed) "))
+year = int(input("What year do you want? "))
+
+EndTwo = str(year)
+EndTwo = EndTwo[2:4]
+
+Century = int(year / 100)
+
+anchor = 5 * (Century % 4) % 7 + 2
+
+A = int(int(EndTwo) / 12)
+
+B = int(EndTwo) % 12
+
+C = int(B / 4)
+
+Doomsday = ((A + B + C) % 7) + anchor
+
+Doomsday = Doomsday % 7
+```
+
+Now we need the script to find the doomsday that’s in the month chosen. But the script doesn't know what a doomsday is, so we will create a 2D list which contains the doomsday for every month. We will put the list just after the inputs to keep the script structured. The list will be written as:
+```python
+DoomsdayList = [[4, 4], [6, 6], [8, 8], [10, 10], [12, 12], [9, 5], [5, 9], [14, 3], [7, 11], [11, 7], [4, 1], [29, 2]]
+```
+
+So now the script knows what the doomsdays are, we can make a loop that loops for how many items are in the list, 12, then if `month` is the same as `a` (from the loop) but as it is a 2D list we also need to tell it to look in the second column, so it would look like this:
```

Comment:
```suggestion
Now that the script knows what the doomsdays are, loop through all 12 items in the list. Then we can check if `month` is the same as `a` (declared in the loop). However, since it is a 2D list, we also need to tell it to look in the second column. As a result, the code looks like this:
```